kover/kover-cli/src/main/kotlin/kotlinx/kover/cli/commands/ReportCommand.kt

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kotlinx.kover.cli.commands
+
+import com.intellij.rt.coverage.report.ReportLoadStrategy
+import com.intellij.rt.coverage.report.ReportLoadStrategy.RawReportLoadStrategy
+import com.intellij.rt.coverage.report.Reporter
+import com.intellij.rt.coverage.report.data.BinaryReport
+import com.intellij.rt.coverage.report.data.Filters
+import com.intellij.rt.coverage.report.data.Module
+import kotlinx.kover.cli.util.asPatterns
+import org.kohsuke.args4j.Argument
+import org.kohsuke.args4j.Option
+import java.io.File
+import java.io.IOException
+import java.io.PrintWriter
+
+
+internal class ReportCommand : Command {
+    @Argument(usage = "list of raw reports files", metaVar = "<raw-report-path>")
+    private var rawReports: MutableList<File> = ArrayList()
+
+    @Option(name = "--src", usage = "location of the source files root", metaVar = "<sources-path>", required = true)
+    private var sourceRoots: MutableList<File> = ArrayList()
+
+    @Option(
+        name = "--classfiles",
+        usage = "location of the compiled class-files root (must be original and not instrumented)",
+        metaVar = "<class-file-path>",
+        required = true
+    )
+    private var outputRoots: MutableList<File> = ArrayList()
+
+    @Option(name = "--xml", usage = "generate a XML report in the specified path", metaVar = "<xml-file-path>")
+    private var xmlFile: File? = null
+
+    @Option(name = "--html", usage = "generate a HTML report in the specified path", metaVar = "<html-dir>")
+    private var htmlDir: File? = null
+
+    @Option(name = "--title", usage = "title in the HTML report", metaVar = "<html-title>")
+    private var htmlTitle: String? = null
+
+    @Option(
+        name = "--include",
+        usage = "filter to include classes, wildcards `*` and `?` are acceptable",
+        metaVar = "<class-name>"
+    )
+    private var includeClasses: MutableList<String> = ArrayList()
+
+    @Option(
+        name = "--exclude",
+        usage = "filter to exclude classes, wildcards `*` and `?` are acceptable",
+        metaVar = "<class-name>"
+    )
+    private var excludeClasses: MutableList<String> = ArrayList()
+
+    @Option(
+        name = "--excludeAnnotation",
+        usage = "filter to include classes and functions marked by this annotation, wildcards `*` and `?` are acceptable",
+        metaVar = "<annotation-name>"
+    )
+    private var excludeAnnotation: MutableList<String> = ArrayList()
+
+    override val name: String = "report"
+
+    override val description: String = "Generates human-readable reports in various formats from raw report files (*.ic)"
```

Comment:
In this repository, `.ic` files are named 'binary' reports, while 'raw' reports/format is related only to the results of the offline instrumented coverage run.